### ISSUE TITLE
Import data from cytoscape sif and graphml

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#288]
Related to issue #287, but importing is a different task.

**Comment by kdahlquist** [comment]
We decided that the File > Open menu item will refer to the native .xlsx format and that we will have a separate menu item File > Import > Cytoscape SIF, etc.

**Comment by kdahlquist** [comment]
Tasks for @kdahlquist:
* Convert demo files 1-4 to SIF, GraphML, GML.
* Once converted, load into Cytoscape

**Comment by dondi** [comment]
The v1.15 beta on the website now implements SIF import also. This gives SIF a full-cycle implementation now, allowing both import and export.

To save time, I think we'll have to stay light on the detail with error checking for imports. I implemented a generalized import error dialog in case something comes up, but it will take more effort to go to the level of reporting seen in the native .xlsx opening.

That said, upon looking at this issue, I observed that many of the error checks we perform on .xlsx would apply to any other imported file, regardless of the format. However, applying those checks without redundantly copying code will take some refactoring. I will write that up as a separate issue, likely feasible only after we have merged the current `beta` branch into `master`.

**Comment by dondi** [comment]
That refactoring issue has been written up as #300.

**Comment by dondi** [comment]
Current work-in-progress is import from GraphML. This one will require some documentation when we are done because GraphML captures a superset of what GRNsight can display. My current intent is to allow the successful import of any GraphML file, but the import will ignore features that GRNsight does not support (e.g., "hyperedges," nested graphs). Again my point of reference is http://graphml.graphdrawing.org/primer/graphml-primer.html. I aim to have GRNsight successfully import all of the sample _.graphml_ files that are posted there, ignoring the portions of those files that do not have an equivalent in GRNsight.

In summary, the intent is to support importing of the following GraphML entities:
- nodes (ID'ed by gene/node name)
- edges (directed only)
- edge weights
All other features and attributes will be ignored (for now).

**Comment by dondi** [comment]
OK, GraphML import is now live on the beta 1.15 page. I was on that track already so I decided to finish that first, prior to looking at the testing results from the pre-existing functionality.

**Comment by kdahlquist** [comment]
We are not going to import from GML right now so that we can prioritize other issues.

**Comment by kdahlquist** [comment]
I am going to close this because the basic implementation is done and more discussion has been shifted over to #309 and #314.
